pkg/protocol/commitment_verifier.go

```diff
@@ -91,8 +91,8 @@ func (c *CommitmentVerifier) verifyCommitment(commitment *model.Commitment, atte
 	//    than it actually is. Nodes might consider to switch to this chain, even though it is invalid which will be discovered
 	//    before the candidate chain/engine is activated (it will never get heavier than the current chain).
 	c.cumulativeWeight += seatCount
-	if c.cumulativeWeight <= commitment.CumulativeWeight() {
-		return nil, 0, ierrors.Errorf("invalid cumulative weight for commitment %s", commitment.ID())
+	if c.cumulativeWeight > commitment.CumulativeWeight() {
```

Comment:
The comment above is not true for this comparison anymore
> 	// 3. Verify that cumulative weight of commitment matches (as an upper bound) with calculated weight from attestations.
I would explain why you check the cumulative weight of the commitment as a lower bound now.